src/services/getAchievementData.ts

```diff
@@ -1,7 +1,7 @@
 import {Achievement} from "@/entities/Achievement";
 import {Session} from "next-auth";
 
-export default async function getAchievementData(args: {appId: string}, session: Session | null): Promise<Array<Achievement>> {
+export default async function getAchievementData(args: {appId: string}, session: Session): Promise<Array<Achievement>> {
```

Comment:
This requires callers to check `status = authenticated`, which fixes a lot of strange behavior